framework/src/st_things/things_stack/framework/things_data_manager.c

```diff
@@ -754,395 +714,240 @@ int parse_things_cloud_json(const char *filename)
 	return ret;
 }
 
-static int parse_things_info_json(const char *filename)
+static int parse_configuration_json(cJSON *configuration)
 {
+	int ret = 0;
+	if (configuration != NULL) {
+		int connectivity_type = 0;
+
+		cJSON *easysetup = cJSON_GetObjectItem(configuration, KEY_CONFIGURATION_EASYSETUP);
+		if (easysetup != NULL) {
+			cJSON *connectivity = cJSON_GetObjectItem(easysetup, KEY_CONFIGURATION_EASYSETUP_CONNECTIVITY);
+			if (connectivity != NULL) {
+				cJSON *type = cJSON_GetObjectItem(connectivity, KEY_CONFIGURATION_EASYSETUP_CONNECTIVITY_TYPE);
+				connectivity_type = type->valueint;
+				THINGS_LOG_V(TAG, "[configuration] type       : %d", connectivity_type);
+				if (connectivity_type == 1) {
+					es_conn_type = es_conn_type_softap;
+					cJSON *softap = cJSON_GetObjectItem(connectivity, KEY_CONFIGURATION_EASYSETUP_CONNECTIVITY_SOFTAP);
+					if (softap != NULL) {
+						cJSON *setup_id = cJSON_GetObjectItem(softap, KEY_CONFIGURATION_EASYSETUP_CONNECTIVITY_SOFTAP_SETUPID);
+						cJSON *artik = cJSON_GetObjectItem(softap, KEY_CONFIGURATION_EASYSETUP_CONNECTIVITY_SOFTAP_ARTIK);
+
+						if (setup_id != NULL) {
+							if (strlen(setup_id->valuestring) != 3) {
+								THINGS_LOG_V(TAG, "setup_id exceeds 3 bytes. please check (3 bytes are fixed sizes.)");
+								goto JSON_ERROR;
+							}
+							is_artik = false;
+							if (artik->type == cJSON_True) {
+								is_artik = true;
+							}
 
```

Comment:
Logic is fine.
Reducing the complexity of nested conditions are recommended, How about changing the sequence of condition and reduce the depth of condition?